src/js/protocols/stm32.js

```diff
@@ -114,10 +116,14 @@ STM32_protocol.prototype.connect = function (port, baud, hex, options, callback)
 
         const onDisconnect = disconnectionResult => {
             if (disconnectionResult) {
+                GUI.log('Disconnected, waiting 5s for DFU connection');
+                console.log(`Disconnected, waiting 5s for DFU connection`);
                 // delay to allow board to boot in bootloader mode
                 // required to detect if a DFU device appears
-                setTimeout(startFlashing, 1000);
+                setTimeout(startFlashing, 5000);
```

Comment:
Wouldn't it be better to periodically check for the presence of a DFU connection, rather than just waiting for a specified time after disconnecting from the GUI, then running `startFlashing`without knowing that a DFU connection existed?

I assume that on Windows the serial disconnection is much faster, but on a Mac it really can take 5s.

Reply:
The expert in this is @haslinghuis that has fight a lot against it. But I suppose the correct is to wait to the DFU, but with a timeout to "fail" if it does not appear in a reasonable amount of time.